src/kafkit/registry/serializer.py

```diff
@@ -162,13 +185,17 @@ def __init__(self, *, registry):
             schema = await self._registry.get_schema_by_id(schema_id)
         elif schema is not None:
             schema_id = await self._registry.register_schema(
-                schema=schema, subject=subject)
+                schema=schema, subject=subject
+            )
         else:
-            raise RuntimeError('Pass either a schema or schema_id parameter.')
+            raise RuntimeError("Pass either a schema or schema_id parameter.")
+        assert isinstance(schema, dict)  # to guide mypy
```

Comment:
In this case, you're just unconfusing mypy about whether `schema` is `None`, so I would simplify this to `assert schema`. The `isinstance` makes me think something more complicated is going on than it actually is.

Another option that avoids the `assert` (but which may seem awkward; it's a matter of taste) to replace the `else` clause with:
```
if not schema:
    raise RuntimeError("Pass either a schema or schema_id parameter")
```